test/opt/opextinst_forward_ref_fixup_pass_test.cpp

```diff
@@ -0,0 +1,365 @@
+          %6 = OpLabel
+               OpReturn
+               OpFunctionEnd;
+  )";
+  const auto result =
+      SinglePassRunAndMatch<OpExtInstWithForwardReferenceFixupPass>(
+          kTest, /* do_validation= */ true);
+  EXPECT_EQ(std::get<1>(result), Pass::Status::SuccessWithoutChange);
+}
+
+TEST_F(OpExtInstForwardRefFixupPassTest, NoForwardRef_NoChange) {
+  const std::string kTest = R"(
+                   OpCapability Shader
+                   OpExtension "SPV_KHR_non_semantic_info"
+              %1 = OpExtInstImport "NonSemantic.Shader.DebugInfo.100"
+                   OpMemoryModel Logical GLSL450
+                   OpEntryPoint GLCompute %main "main"
+                   OpExecutionMode %main LocalSize 1 1 1
+              %3 = OpString "/usr/local/google/home/nathangauer/projects/DirectXShaderCompiler/repro.hlsl"
+              %4 = OpString "// RUN: %dxc -T cs_6_0 %s -E main -spirv -fspv-target-env=vulkan1.1 -fspv-debug=vulkan-with-source | FileCheck %s
+
+[numthreads(1, 1, 1)]
+void main() {
+}
+"
+              %5 = OpString "main"
+              %6 = OpString ""
+              %7 = OpString "3f3d3740"
+              %8 = OpString " -E main -T cs_6_0 -spirv -fspv-target-env=vulkan1.1 -fspv-debug=vulkan-with-source -Qembed_debug"
+                   OpName %main "main"
+           %void = OpTypeVoid
+           %uint = OpTypeInt 32 0
+         %uint_3 = OpConstant %uint 3
+         %uint_1 = OpConstant %uint 1
+         %uint_4 = OpConstant %uint 4
+         %uint_5 = OpConstant %uint 5
+             %20 = OpTypeFunction %void
+             %10 = OpExtInst %void %1 DebugTypeFunction %uint_3 %void
+             %12 = OpExtInst %void %1 DebugSource %3 %4
+             %13 = OpExtInst %void %1 DebugCompilationUnit %uint_1 %uint_4 %12 %uint_5
+             %17 = OpExtInst %void %1 DebugFunction %5 %10 %12 %uint_4 %uint_1 %13 %6 %uint_3 %uint_4
+             %18 = OpExtInst %void %1 DebugEntryPoint %17 %13 %7 %8
+;  CHECK: {{.*}} = OpExtInst %void %1 DebugTypeFunction %uint_3 %void
+;  CHECK: {{.*}} = OpExtInst %void %1 DebugSource {{.*}} {{.*}}
+;  CHECK: {{.*}} = OpExtInst %void %1 DebugCompilationUnit %uint_1 %uint_4 {{.*}} %uint_5
+;  CHECK: {{.*}} = OpExtInst %void %1 DebugFunction {{.*}} {{.*}} {{.*}} %uint_4 %uint_1 {{.*}} {{.*}} %uint_3 %uint_4
+;  CHECK: {{.*}} = OpExtInst %void %1 DebugEntryPoint {{.*}} {{.*}} {{.*}} {{.*}}
+           %main = OpFunction %void None %20
+             %21 = OpLabel
+             %22 = OpExtInst %void %1 DebugFunctionDefinition %17 %main
+             %23 = OpExtInst %void %1 DebugLine %12 %uint_5 %uint_5 %uint_1 %uint_1
+; CHECK:  {{.*}} = OpExtInst %void %1 DebugFunctionDefinition {{.*}} %main
+; CHECK:  {{.*}} = OpExtInst %void %1 DebugLine {{.*}} %uint_5 %uint_5 %uint_1 %uint_1
+                   OpReturn
+                   OpFunctionEnd
+  )";
+  const auto result =
+      SinglePassRunAndMatch<OpExtInstWithForwardReferenceFixupPass>(
+          kTest, /* do_validation= */ true);
```

Comment:
If you want to test that there is no change, you can do something like this:

https://github.com/KhronosGroup/SPIRV-Tools/blob/65d30c31508e47d34f7698f511aea4800e3d44fe/test/opt/redundancy_elimination_test.cpp#L359

Reply:
Ahh that's better, changed thanks.